pkg/tlsutil/primatives.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tlsutil
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math"
+	"math/big"
+	"time"
+
+	"k8s.io/api/core/v1"
+)
+
+const (
+	rsaKeySize   = 2048
+	duration365d = time.Hour * 24 * 365
+)
+
+// NewPrivateKey returns randomly generated RSA private key.
+func NewPrivateKey() (*rsa.PrivateKey, error) {
```

Comment:
Do all of these helpers need to be exported? I can't see how they will end up being used by GenerateCert() but I assume it's going to be in the same package. 
So we can keep these functions as lowercase and unexported.
They're not meant to be exposed to the user.

Reply:
yeah, you are right. they are not meant to be exported.